data_steward/cdr_cleaner/cleaning_rules/create_person_ext_table.py

```diff
@@ -0,0 +1,129 @@
+state_of_residence_source_value: the concept_name from the concept table for the state_of_residence_concept_id
+person_id (as research_id) can be pulled from the person table
+"""
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.bq_utils import WRITE_TRUNCATE
+from utils.bq import JINJA_ENV
+
+# Query to create person_ext table
+PERSON_EXT_TABLE_QUERY = JINJA_ENV.from_string("""
+SELECT p.person_id, e.src_id,
+o.value_source_concept_id AS state_of_residence_concept_id,
+c.concept_name AS state_of_residence_source_value
+FROM `{{project}}.{{dataset}}.person` p
+LEFT JOIN `{{project}}.{{dataset}}.observation` o
+ON p.person_id = o.person_id AND o.observation_source_concept_id = 1585249
+LEFT JOIN `{{project}}.{{dataset}}.concept` c
+ON o.value_source_concept_id = c.concept_id AND o.observation_source_concept_id = 1585249
+LEFT JOIN  `{{project}}.{{dataset}}.observation_ext` e
+ON o.observation_id = e.observation_id AND o.observation_source_concept_id = 1585249
+""")
+
+tables = ['person_ext']
+
+
+class CreatePersonExtTable(BaseCleaningRule):
+    """
+    Create person_ext table after DEID, adds three non-standard fields:
+    state_of_residence_concept_id, state_of_residence_source_value
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = ('Create person_ext table')
+        super().__init__(issue_numbers=['DC1012'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.DEID_CLEAN],
+                         affected_tables=tables,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        query_list = []
+        for table in tables:
+            query_list.append({
+                cdr_consts.QUERY:
+                    PERSON_EXT_TABLE_QUERY.render(project=self.project_id,
+                                                  dataset=self.dataset_id),
+                cdr_consts.DESTINATION_TABLE:
+                    table,
+                cdr_consts.DESTINATION_DATASET:
+                    self.dataset_id,
+                cdr_consts.DISPOSITION:
+                    WRITE_TRUNCATE
```

Comment:
If you were to change the query to a DDL statement, (like `CREATE OR REPLACE TABLE...`), you could drop these parameters entirely.

Reply:
I thought we weren't supposed to use DDL statements? Is this not the case anymore?

Reply:
We still aren't using DML statements frequently (`update`, `delete`), but DDL statements are ok (create a table from a result set).  This is an optional suggestion and is not a required change.

Reply:
ahh was confusing DML vs DDL!